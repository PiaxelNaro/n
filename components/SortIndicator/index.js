// @flow

import React from 'react';
import styled from 'styled-components';
import FaSort from 'react-icons/lib/fa/sort';
import FaSortAsc from 'react-icons/lib/fa/sort-asc';
import FaSortDesc from 'react-icons/lib/fa/sort-desc';

const SIZE = 13;
const SORT_ICONS = {
    asc: <FaSortAsc size={SIZE} color="currentColor" />,
    desc: <FaSortDesc size={SIZE} color="currentColor" />,
};
const DEFAULT = <FaSort size={SIZE} color="#CCC" />;

const Wrapper = styled.div`
    position: relative;
    display: block;
    width: ${SIZE}px;
    height: ${SIZE}px;

    > svg {
        position: absolute;
    }
`;

const SortIndicator = ({ direction }: { direction?: 'asc' | 'desc' | null }) => (
    <Wrapper>
        {DEFAULT}
        {direction ? SORT_ICONS[direction] : null}
    </Wrapper>
);

export default SortIndicator;
