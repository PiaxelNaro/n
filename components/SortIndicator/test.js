// @flow
import React from 'react';
import { shallow } from 'enzyme';
import SortIndicator from '.';

describe('<SortIndicator />', () => {
    it('render sort direction', () => {
        expect(shallow(<SortIndicator />)).toMatchSnapshot();
        expect(shallow(<SortIndicator direction="asc" />)).toMatchSnapshot();
        expect(shallow(<SortIndicator direction="desc" />)).toMatchSnapshot();
    });
});
