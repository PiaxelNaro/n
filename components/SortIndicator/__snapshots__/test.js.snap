// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortIndicator /> render sort direction 1`] = `
<SortIndicator__Wrapper>
  <FaSort
    color="#CCC"
    size={13}
  />
</SortIndicator__Wrapper>
`;

exports[`<SortIndicator /> render sort direction 2`] = `
<SortIndicator__Wrapper>
  <FaSort
    color="#CCC"
    size={13}
  />
  <FaSortAsc
    color="currentColor"
    size={13}
  />
</SortIndicator__Wrapper>
`;

exports[`<SortIndicator /> render sort direction 3`] = `
<SortIndicator__Wrapper>
  <FaSort
    color="#CCC"
    size={13}
  />
  <FaSortDesc
    color="currentColor"
    size={13}
  />
</SortIndicator__Wrapper>
`;
